### ISSUE TITLE
Emit JavaScript Cached modules like `esm.sh`

### ISSUE THREAD

**Issue by ericlery** [jsr-io/jsr#133]
At this moment, this code doesn't work. JSR don't have the ability to emit JavaScript and cache it to serve browsers.

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>JSR in the Browsers</title>
</head>
<body>
  <script type="module">
    import { yassify} from "https:/jsr.io/@kwhinnery/yassify@1.0.0";

    alert(yassify("In the browser!"));
  </script>
</body>
</html>

```

Although doing similar with `esh.sh` works:
```html
<script type="module">
  import { yassify} from "https://esm.sh/jsr/@kwhinnery/yassify/1.0.0";

  alert(yassify("In the browser!"));
</script>
```

**JSR should implement emitting and caching for browsers.** It would be the ultimate JavaScript registry.

**Comment by ije** [comment]
hi esm.sh maintainer here, i guess you can use esm.sh for now, util this feature was been added

```diff
- https://esm.sh/jsr/@kwhinnery/yassify@1.0.0
+ https://jsr.io/@kwhinnery/yassify@1.0.0
```
if you have problem with esm.sh/jsr welcome to file issue [here](https://github.com/esm-dev/esm.sh/issues)

**Comment by jespertheend** [comment]
The JSR website says it works with any runtime. Which is a pretty bold claim when it doesn't work in browsers :P

**Comment by ericlery** [comment]
> The JSR website says it works with any runtime. Which is a pretty bold claim when it doesn't work in browsers :P

It's actually a bit more complicated than that. It works on projects when your back end process/bundle your front end.

But for me personally, the good ol' browser link is the way for quick and dirty prototyping. For some people, it's even their norm.

**Comment by jespertheend** [comment]
For me it's my norm, it's why I exclusively write .js files 😅 

Fwiw, [jsr's usage policy](https://jsr.io/docs/usage-policy#unacceptable-use) says:

> It is not acceptable to use JSR as a CDN for serving assets directly to web applications in a browser, except for development purposes. JSR is not a general purpose file hosting service.

So I suppose that even if this does get added, it will still be pretty useless.

I think maybe instead we should start bugging jsDelivr. They already support npm, so hopefully they could make use of the npm compatibility layer without too much effort. [Their site says](https://www.jsdelivr.com/documentation#id-custom-cdn-hosting):

> ### Custom CDN Hosting
> We can work together and set up a custom configuration for your project. This way, you can have full control over your files and the ability to utilize the full power of jsDelivr.

**Comment by marvinhagemeister** [comment]
FYI esm.sh already added support for JSR.

**Comment by jespertheend** [comment]
Ah cool! I wasn't aware.

**Comment by ericlery** [comment]
Yes to all of you that **_re_**-mention it, esm.sh already added support of JSR, that's the title and the whole point for this issue: 
> Make it available for JSR **_too_** please.

Why should we use something else to get JSR modules?

**Comment by mhanuszh** [comment]
I'm not here to annoy anyone subscribed to this issue, but...
Is this feature being even considered by the devs? I haven't found anything about them planning on adding this.

**Comment by dbushell** [comment]
> Let us be clear: **Deno is not removing HTTP imports.** 
> https://deno.com/blog/http-imports

Deno is supposedly committed to supporting HTTP imports. Or at least not removing them.

With the Deno standard library now published exclusively on JSR — [deno.land/std](https://deno.land/std) is months out of date, and [importing from source](https://github.com/jsr-io/jsr/issues/679) is broken — I'd expect JSR to surface some kind of HTTP import URL.

**Comment by dirkluijk** [comment]
Surprised that this is indeed not supported. I suppose it would be quite easy to serve packages directly to browsers, and seems like a missed opportunity. 😉

**Comment by ericlery** [comment]
> Surprised that this is not (yet?) supported. I suppose it would be quite easy to serve packages directly to browsers, and seems like a missed opportunity. 😉

Absolutely. I can't believe I've opened this issue nearly one year ago!

**Comment by BlackAsLight** [comment]
If I was to guess, they most likely don't want to support it at least at this time due to the extra hosting cost that would burden JSR with.

**Comment by ericlery** [comment]
> If I was to guess, they most likely don't want to support it at least at this time due to the extra hosting cost that would burden JSR with.

Exactly. I hope some Tech Giant will stand up to support JSR. Maybe one that is part of WinterCG like Vercel or even better Cloudflare.

**Comment by scarf005** [comment]
![Image](https://github.com/user-attachments/assets/faea669a-27bd-4aa9-842b-e57d21effe68)

not supporting browser makes the claim "browser compatible" moot. it should list `Browser` in option with link to esm.sh at least.
